From 17d90e52e77fc92237025e2329916cc2d013aa24 Mon Sep 17 00:00:00 2001
From: Gustavo Padovan <gustavo.padovan@collabora.co.uk>
Date: Tue, 31 May 2016 16:59:11 -0300
Subject: [PATCH 59/88] staging/android: remove drv_name from sync_timeline
Git-commit: b9bc2b7b658a4bee8884e2c339a0a27f4bfa7f22
Patch-mainline: v4.8-rc1
References: fate#321223

As it is internal to sw_sync now this value will always be "sw_sync".

Signed-off-by: Gustavo Padovan <gustavo.padovan@collabora.co.uk>
Reviewed-by: Daniel Vetter <daniel.vetter@ffwll.ch>
Reviewed-by: Sumit Semwal <sumit.semwal@linaro.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
Signed-off-by: Patrik Jakobsson <pjakobsson@suse.de>
---
 drivers/staging/android/sw_sync.c    | 11 +++--------
 drivers/staging/android/sync.h       |  2 --
 drivers/staging/android/sync_debug.c |  2 +-
 3 files changed, 4 insertions(+), 11 deletions(-)

diff --git a/drivers/staging/android/sw_sync.c b/drivers/staging/android/sw_sync.c
index cf4de27266a4..ae56eceeeaac 100644
--- a/drivers/staging/android/sw_sync.c
+++ b/drivers/staging/android/sw_sync.c
@@ -48,14 +48,12 @@ static inline struct sync_pt *fence_to_sync_pt(struct fence *fence)
 
 /**
  * sync_timeline_create() - creates a sync object
- * @drv_name:	sync_timeline driver name
  * @name:	sync_timeline name
  *
  * Creates a new sync_timeline. Returns the sync_timeline object or NULL in
  * case of error.
  */
-struct sync_timeline *sync_timeline_create(const char *drv_name,
-					   const char *name)
+struct sync_timeline *sync_timeline_create(const char *name)
 {
 	struct sync_timeline *obj;
 
@@ -66,7 +64,6 @@ struct sync_timeline *sync_timeline_create(const char *drv_name,
 	kref_init(&obj->kref);
 	obj->context = fence_context_alloc(1);
 	strlcpy(obj->name, name, sizeof(obj->name));
-	strlcpy(obj->drv_name, drv_name, sizeof(obj->drv_name));
 
 	INIT_LIST_HEAD(&obj->child_list_head);
 	INIT_LIST_HEAD(&obj->active_list_head);
@@ -161,9 +158,7 @@ static struct sync_pt *sync_pt_create(struct sync_timeline *obj, int size,
 
 static const char *timeline_fence_get_driver_name(struct fence *fence)
 {
-	struct sync_timeline *parent = fence_parent(fence);
-
-	return parent->drv_name;
+	return "sw_sync";
 }
 
 static const char *timeline_fence_get_timeline_name(struct fence *fence)
@@ -247,7 +242,7 @@ static int sw_sync_debugfs_open(struct inode *inode, struct file *file)
 
 	get_task_comm(task_comm, current);
 
-	obj = sync_timeline_create("sw_sync", task_comm);
+	obj = sync_timeline_create(task_comm);
 	if (!obj)
 		return -ENOMEM;
 
diff --git a/drivers/staging/android/sync.h b/drivers/staging/android/sync.h
index 3c551f5401ee..48e2d1c8c3ef 100644
--- a/drivers/staging/android/sync.h
+++ b/drivers/staging/android/sync.h
@@ -23,7 +23,6 @@
 /**
  * struct sync_timeline - sync object
  * @kref:		reference count on fence.
- * @drv_name:		drv_name of the driver using the sync_timeline
  * @name:		name of the sync_timeline. Useful for debugging
  * @child_list_head:	list of children sync_pts for this sync_timeline
  * @child_list_lock:	lock protecting @child_list_head and fence.status
@@ -32,7 +31,6 @@
  */
 struct sync_timeline {
 	struct kref		kref;
-	char			drv_name[32];
 	char			name[32];
 
 	/* protected by child_list_lock */
diff --git a/drivers/staging/android/sync_debug.c b/drivers/staging/android/sync_debug.c
index 77386d2ec437..b2697a12b424 100644
--- a/drivers/staging/android/sync_debug.c
+++ b/drivers/staging/android/sync_debug.c
@@ -120,7 +120,7 @@ static void sync_print_obj(struct seq_file *s, struct sync_timeline *obj)
 	struct list_head *pos;
 	unsigned long flags;
 
-	seq_printf(s, "%s %s: %d\n", obj->name, obj->drv_name, obj->value);
+	seq_printf(s, "%s: %d\n", obj->name, obj->value);
 
 	spin_lock_irqsave(&obj->child_list_lock, flags);
 	list_for_each(pos, &obj->child_list_head) {
-- 
2.12.0
